### PR TITLE
clang 14 support

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

Clang 14 compilation

### Motivation

Latest toolchain on ubuntu 22.04

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/ebpf/c/dentry_resolver.h [hunk 1]
```diff
@@ -173,9 +185,9 @@ int __attribute__((always_inline)) resolve_dentry_tail_call(void *ctx, struct de
     }
 
     if (input->iteration == DR_MAX_TAIL_CALL) {
-        map_value.name[0] = 0;
-        map_value.parent.mount_id = 0;
-        map_value.parent.ino = 0;
+        map_value->name[0] = 0;
+        map_value->parent.mount_id = 0;
+        map_value->parent.ino = 0;
         bpf_map_update_elem(&pathnames, &next_key, &map_value, BPF_ANY);
```
Comment (paulcacheux) [inline pkg/security/ebpf/c/dentry_resolver.h:191]: shouldn't this be ?
```suggestion
        bpf_map_update_elem(&pathnames, &next_key, map_value, BPF_ANY);
```
also I don't think you can update_elem with a value coming from another maps on 4.14

**Comment by paulcacheux** [comment]
How much is clang-14 a priority for you ? Is CWS the only remaining blocker ?

File: pkg/security/ebpf/c/dentry_resolver.h (hunk 1 above)
Comment (nplanel) [reply pkg/security/ebpf/c/dentry_resolver.h]: good catch, but why the CI pass ?

**Comment by nplanel** [comment]
> How much is clang-14 a priority for you ? Is CWS the only remaining blocker ?

I upgrade my laptop to 22.04 and this issue came, so here the fix, it's not blocking or urgent

The previous patch (alternative fix) was to reduce the stack size : https://github.com/DataDog/datadog-agent/pull/13210/commits/1b9ab00781e34ce2b0f4bee5cb91f6426a360146#diff-54f316ac5735fec92aaa1384ac40f89f76d4e2e11fc9a35685b6a3e89c09f74bR20

There are only this issue with CWS

File: pkg/security/ebpf/c/dentry_resolver.h (hunk 1 above)
Comment (paulcacheux) [reply pkg/security/ebpf/c/dentry_resolver.h]: I don't think it does https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/165040117

Comment (nplanel) [reply pkg/security/ebpf/c/dentry_resolver.h]: ah you are in allow to failed ...
yeah map value to map is in 4.18 https://github.com/torvalds/linux/commit/d71962f3e627b5941804036755c844fabfb65ff5

Comment (nplanel) [reply pkg/security/ebpf/c/dentry_resolver.h:191]: so I push the previous solution but it's reducing by 10 chars your path
Is it acceptable ?

Comment (paulcacheux) [reply pkg/security/ebpf/c/dentry_resolver.h:191]: not really no, let me get back to you on what could work

**Comment by nplanel** [comment]
latest version here https://github.com/DataDog/datadog-agent/pull/13255

**Pull request closed without merging**